src/elm/Pile.elm

```diff
@@ -0,0 +1,240 @@
+module Pile exposing (..)
+
+import AssocList
+import Data
+import Draggable
+import Draggable.Events
+import Html
+import Html.Attributes
+import Html.Events
+import Math.Vector2
+
+
+type Data
+    = Post Data.Post
+    | Image Data.Image
+
+
+type alias Id =
+    String
+
+
+type alias Card =
+    { id : Id
+    , position : Math.Vector2.Vec2
+    , cont : Data
+    }
+
+
+dragCardBy : Math.Vector2.Vec2 -> Card -> Card
+dragCardBy delta card =
+    { card | position = card.position |> Math.Vector2.add delta }
+
+
+type alias CardPile =
+    { uid : Int
+    , activeCard : Maybe Card
+    , idleCards : List Card
+    }
+
+
+emptyPile : CardPile
+emptyPile =
+    CardPile 0 Nothing []
+
+
+addCard : ( Math.Vector2.Vec2, Data ) -> CardPile -> CardPile
+addCard ( position, content ) ({ uid, idleCards } as pile) =
+    { pile
+        | uid = uid + 1
+        , idleCards = Card (String.fromInt uid) position content :: idleCards
+    }
+
+
+makeCardPile : List ( Math.Vector2.Vec2, Data ) -> CardPile
+makeCardPile elements =
+    List.foldl addCard emptyPile elements
+
+
+allCards : CardPile -> List Card
+allCards { activeCard, idleCards } =
+    activeCard
+        |> Maybe.map (\c -> c :: idleCards)
+        |> Maybe.withDefault idleCards
+
+
+startDragging : Id -> CardPile -> CardPile
+startDragging id pile =
+    let
+        ( targetAsList, others ) =
+            List.partition (\card -> card.id == id) pile.idleCards
+    in
+    { pile
+        | idleCards = others
+        , activeCard = targetAsList |> List.head
+    }
+
+
+stopDragging : CardPile -> CardPile
+stopDragging pack =
+    { pack | idleCards = allCards pack, activeCard = Nothing }
+
+
+dragActiveBy : Math.Vector2.Vec2 -> CardPile -> CardPile
+dragActiveBy delta ({ activeCard } as pack) =
+    { pack | activeCard = activeCard |> Maybe.map (dragCardBy delta) }
+
+
+type alias Model =
+    { activePile : Maybe Data.SectionId
+    , piles : AssocList.Dict Data.SectionId CardPile
```

Comment:
Not sure what diff is between `AssocList.Dict` and core `DIct` since I've never needed to use, but if this works, fine.

Reply:
It lets us use any type as the key while sacrificing lookup speed because keys don't have to be comparable. But we have 3 entries in the Dict so it's a good deal.